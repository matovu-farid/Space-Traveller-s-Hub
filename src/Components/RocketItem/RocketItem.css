.rocket-image img {
  width: 150px;
  height: 150px;
  object-fit: fill;
  

}

.rocket-image {
  margin-right: 10px;
}

.rocket-item {
  display: flex;
  
  margin: 5px 0;

}



.rocket-item button {
  background-color: var(--accent-color);
  font-size: 0.8rem;
  color: white;
}

.rocket-item .description {
  font-size: 0.8rem;
}

.rocket-item .name {
  margin: 0;
  font-weight: bold;
}

button.reserved {
  background-color: white;
  color: grey;
  border: 1.5px solid grey;
}
