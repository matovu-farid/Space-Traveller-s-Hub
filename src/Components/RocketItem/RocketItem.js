import './RocketItem.css';
import propTypes from 'prop-types';

import { useDispatch } from 'react-redux';
import { cancelReservedRocket, reserveRocket } from '../../redux/rockets/rockets';

const RocketItem = ({
  name, description, image, reserved, id,
}) => {
  const dispatch = useDispatch();

  return (
    <li className="rocket-item">
      <div className="rocket-image">
        <img src={image} alt={name} />
      </div>
      <div className="rocket-info">

        <p className="name">{name}</p>
        <p className="description">{description}</p>
        {(reserved) ? <button className="reserved" onClick={() => dispatch(cancelReservedRocket(id))} type="button">Cancel reservation</button>
          : <button type="button" onClick={() => dispatch(reserveRocket(id))}>Reserve Rocket</button>}

      </div>
    </li>
  );
};

RocketItem.propTypes = {
  id: propTypes.number.isRequired,
  name: propTypes.string.isRequired,
  description: propTypes.string.isRequired,
  image: propTypes.string.isRequired,
  reserved: propTypes.bool.isRequired,

};

export default RocketItem;
