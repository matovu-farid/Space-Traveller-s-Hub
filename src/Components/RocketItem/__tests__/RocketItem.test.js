import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import RocketItem from '../RocketItem';
import store from '../../../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>

        <RocketItem
          id={1}
          name="Falcon 1"
          description="Great rocket"
          image="https://imgur.com/DaCfMsj.jpg"
          reserved
        />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
