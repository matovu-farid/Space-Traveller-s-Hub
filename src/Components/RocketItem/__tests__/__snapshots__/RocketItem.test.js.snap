// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="rocket-item"
>
  <div
    className="rocket-image"
  >
    <img
      alt="Falcon 1"
      src="https://imgur.com/DaCfMsj.jpg"
    />
  </div>
  <div
    className="rocket-info"
  >
    <p
      className="name"
    >
      Falcon 1
    </p>
    <p
      className="description"
    >
      <span
        className="reserved-badge"
      >
        Reserved
      </span>
      <span>
        Great rocket
      </span>
    </p>
    <button
      className="reserved"
      onClick={[Function]}
      type="button"
    >
      Cancel reservation
    </button>
  </div>
</li>
`;
